htdocs/modulebuilder/template/myobject_card.php

```diff
@@ -118,9 +118,12 @@
 	$error=0;
 
 	$permissiontoadd = $user->rights->mymodule->write;
-	$permissiontodelete = $user->rights->mymodule->delete || ($permissiontoadd && $object->status == 0);
-	if (empty($backtopage)) $backtopage = dol_buildpath('/mymodule/myobject_card.php',1).'?id='.($id > 0 ? $id : '__ID__');
-	$backurlforlist = dol_buildpath('/mymodule/myobject_list.php',1);
+	$permissiontodelete = $user->rights->mymodule->delete;
```

Comment:
Why changing rules on $permissiontodelete ?
A draft object should always be deletable when you have permission to create.